pl_bolts/losses/giou.py

```diff
@@ -0,0 +1,41 @@
+"""
+Generalized Intersection over Union (GIoU) loss (Rezatofighi et. al)
+"""
+
+import torch
+
+
+def giou_loss(pred: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
+    """
+    Calculates the generalized intersection over union loss.
+
+    It has been proposed in `Generalized Intersection over Union: A Metric and A
+    Loss for Bounding Box Regression <https://arxiv.org/abs/1902.09630>`_.
+
+    Args:
+        pred: batch of prediction bounding boxes with representation
+              [x_min, y_min, x_max, y_max]
+        target: batch of target bounding boxes with representation
+                [x_min, y_min, x_max, y_max]
+
+    Returns:
+        GIoU loss
+    """
+    eps = 1e-6
```

Comment:
shall this go to constants? this would be depending on if we use AMP (float16) right?

Reply:
Ah good point. Should I extract this out as a global constant or as a passed-in argument?

Reply:
well in fact the question is if we really need to use eps as the eps is needed only for zeros decision...